chain/chain/src/chain.rs

```diff
@@ -2357,6 +2357,30 @@ impl Chain {
     pub fn is_chunk_orphan(&self, hash: &CryptoHash) -> bool {
         self.blocks_with_missing_chunks.contains(hash)
     }
+
+    /// Check if can sync with sync_hash
+    pub fn check_sync_hash_validity(&mut self, sync_hash: &CryptoHash) -> Result<bool, Error> {
+        let head = self.head()?;
+        match self.get_block_header(sync_hash) {
+            Ok(sync_header) => {
+                // The Epoch of sync_hash may be either the current one or the previous one
+                if head.epoch_id == *sync_header.epoch_id()
+                    || head.epoch_id == *sync_header.next_epoch_id()
+                {
+                    let prev_hash = sync_header.prev_hash().clone();
+                    if self.runtime_adapter.is_next_block_epoch_start(&prev_hash)? {
+                        debug_assert!(self.get_block(sync_hash).is_ok());
```

Comment:
I think this assert will fail if some peer tries to sync with us while we are also syncing.

Reply:
so what's wrong was on previous iteration then?

Reply:
Good point. I guess I was just making a comment :)

Reply:
Sure! I've updated the code.